File: java/2537.java
class Solution {
    public long countGood(int[] nums, int k) {
        Map<Integer, Integer> m = new HashMap<>();
        long count = 0;
        int s = 0, f = 0, n = nums.length;
        for (int i = 0; i < n; i++) {
            f += m.compute(nums[i], (a, b) -> b == null ? 1 : (b + 1)) - 1;
            int t = s;
            while (f >= k) {
                f -= m.compute(nums[t++], (a, b) -> b - 1);
                if (f < k)
                    count += (n - i + 0L) * (t - s);
            }
            s = t;
        }
        return count;
    }
}

File: python3/2537.py
class Solution:
    def countGood(self, nums: List[int], k: int) -> int:
        m = {}
        count = 0
        s = 0
        f = 0
        n = len(nums)
        for i in range(n):
            l = (m.pop(nums[i]) if nums[i] in m else 0) + 1
            m.setdefault(nums[i], l)
            f += l - 1
            t = s
            while f >= k:
                l = m.pop(nums[t]) - 1
                m.setdefault(nums[t], l)
                f -= l
                t += 1
                if f < k:
                    count += (n - i) * (t - s)
            s = t
        return count
